runtime/src/snapshot_utils.rs

```diff
@@ -1573,11 +1573,13 @@ fn get_io_error(error: &str) -> SnapshotError {
     SnapshotError::Io(IoError::new(ErrorKind::Other, error))
 }
 
+/// slot is Some(slot) if the serialized bank was 'reserialized' into a non-deterministic format
 pub fn verify_snapshot_archive<P, Q, R>(
     snapshot_archive: P,
     snapshots_to_verify: Q,
     storages_to_verify: R,
     archive_format: ArchiveFormat,
+    slot: Option<Slot>,
```

Comment:
Could the `slot` param be its own enum instead? It'd function basically the same as an Option, but then we'd get some additional type safety and ideally it'd be self-describing too. `VerifyBank` was the best I thought of, but not tied to it.

```rust
#[derive(Debug, Copy, Clone)]
/// TODO: comments
pub enum VerifyBank {
    Deterministic,
    NonDeterministic(Slot),
}

pub fn verify_snapshot_archive<P, Q, R>(
    snapshot_archive: P,
    snapshots_to_verify: Q,
    storages_to_verify: R,
    archive_format: ArchiveFormat,
    verify_bank: VerifyBank,
[...]

if let VerifyBank::NonDeterministic(slot) = verify_bank {
[...]
```

Reply:
Other names:
* BankVerification vs BankSnapshotVerification
* BankVerification vs BankVerificationMethod vs BankVerificationStrategy